src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala

```diff
@@ -64,23 +76,39 @@ final case class AuthorizationApiServiceImpl(
     }
   }
 
+  private def getTenantIdOr(
+    selfcareId: UUID
+  )(alternative: => Future[UUID])(implicit contexts: Seq[(String, String)]): Future[UUID] =
+    tenantManagement.getBySelfcareId(selfcareId).map(_.id).recoverWith {
+      case ex if TenantManagementService.is404(ex) => alternative
+    }
+
+  private def upsertTenantBySelfcareId(selfcareId: UUID)(implicit contexts: Seq[(String, String)]): Future[UUID] = for {
+    partyInstitution <- partyProcess.getInstitution(selfcareId)
+    tenantId         <- tenantProcess
+      .selfcareUpsertTenant(partyInstitution.origin, partyInstitution.originId)(partyInstitution.id)
+      .map(_.id)
+  } yield tenantId
+
   def readJwt(identityToken: IdentityToken): Try[(Map[String, AnyRef], String, UUID)] = for {
-    claims         <- jwtReader.getClaims(identityToken.identity_token)
-    sessionClaims  <- extractSessionClaims(claims)
-    organizationId <- getOrganizationId(claims)
-  } yield (sessionClaims, getUserRoles(claims).mkString(","), organizationId)
+    claims        <- jwtReader.getClaims(identityToken.identity_token)
+    sessionClaims <- extractSessionClaims(claims)
+    selfcareId    <- getOrganizationId(claims)
+  } yield (sessionClaims, getUserRoles(claims).mkString(","), selfcareId)
 
   private def extractSessionClaims(claims: JWTClaimsSet): Try[Map[String, AnyRef]] = Try {
     claims.getClaims.asScala.view.filterKeys(admittedSessionClaims.contains).toMap
   }
 
   private def getOrganizationId(claims: JWTClaimsSet): Try[UUID] = for {
```

Comment:
```suggestion
  private def getSelfcareId(claims: JWTClaimsSet): Try[String] = for {
```
`selfcareId` is a `String` 🥲